hazelcast/src/test/java/com/hazelcast/instance/HazelcastInstanceFactoryTest.java

```diff
@@ -214,4 +216,71 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
 
         hazelcastInstance = HazelcastInstanceFactory.newHazelcastInstance(config, randomString(), context);
     }
+
+    @Test
+    public void mobyNameGeneratedIfPropertyEnabled() {
+        Config config = new Config();
+        config.getProperties().put(GroupProperty.MOBY_NAMING_ENABLED.getName(), "true");
+
+        hazelcastInstance = HazelcastInstanceFactory.newHazelcastInstance(config);
+        String name = hazelcastInstance.getName();
+        assertNotNull(name);
+        assertNotContains(name, "_hzInstance_");
+    }
+
+    @Test
+    public void fixedNameGeneratedIfPropertyDisabled() {
+        Config config = new Config();
+        config.getProperties().put(GroupProperty.MOBY_NAMING_ENABLED.getName(), "false");
+
+        hazelcastInstance = HazelcastInstanceFactory.newHazelcastInstance(config);
+        String name = hazelcastInstance.getName();
+        assertNotNull(name);
+        assertContains(name, "_hzInstance_");
+    }
+
+    @Test
+    public void fixedNameGeneratedIfPropertyNotDefined() {
+        Config config = new Config();
+
+        hazelcastInstance = HazelcastInstanceFactory.newHazelcastInstance(config);
+        String name = hazelcastInstance.getName();
+        assertNotNull(name);
+        assertContains(name, "_hzInstance_");
+    }
+
+    @Test
+    public void mobyNameGeneratedIfSystemPropertyEnabled() {
+        Config config = new Config();
+        GroupProperty.MOBY_NAMING_ENABLED.setSystemProperty("true");
+
+        hazelcastInstance = HazelcastInstanceFactory.newHazelcastInstance(config);
+        String name = hazelcastInstance.getName();
+        assertNotNull(name);
+        assertNotContains(name, "_hzInstance_");
+    }
+
+    @Test
+    public void mobyNameUniquenessProvided() {
```

Comment:
Is this test leftover? I dont see any assertions. Also there is printing to stdout

Reply:
It was a check that if we couldn't find an unique name we add a number to the end. But after I added several attempts it is hard to reproduce it in reasonable time so I just removed this test.